views/default/js/elgg/spinner.js

```diff
@@ -0,0 +1,66 @@
+define('elgg/spinner', function (require) {
+	var $ = require('jquery'),
+		elgg = require('elgg'),
+
+		num_running = 0,
+		$spinner;
+
+	var exports = {
+
+		/**
+		 * Add an instance of the loading indicator
+		 *
+		 * @param {jQuery} $wait $-wrapped element(s) to which to add the elgg-wait-cursor class
+		 */
+		add: function ($wait) {
+			if ($wait instanceof $) {
+				$wait.addClass('elgg-wait-cursor');
+			}
+
+			if (!num_running) {
+				if ($spinner) {
+					$spinner.show();
+				} else {
+					$spinner = $('<div class="elgg-spinner elgg-ajax-loader"></div>');
+					$spinner.appendTo('body');
+				}
+			}
+			num_running += 1;
+		},
+
+		/**
+		 * Remove an instance of the spinner. The indicator is displayed until all instances are removed.
+		 *
+		 * @param {jQuery} $wait       $-wrapped element(s) from which to remove the elgg-wait-cursor class
+		 * @param {Object} fadeOptions Options for the $.fadeOut method
+		 */
+		remove: function ($wait, fadeOptions) {
```

Comment:
I could get rid of remove() if I returned a Promise ($.Deferred) that the user would manually resolve().